drivers/user_permission_report/app/controllers/user_permission_report/warehouse_reports/reports_controller.rb

```diff
@@ -6,37 +6,56 @@
 
 module UserPermissionReport::WarehouseReports
   class ReportsController < ApplicationController
-
     before_action :set_group_associations
 
     def index
-      if params[:q].present?
-        @users = User.active.
-          text_search(params[:q]).
-          order(:last_name, :first_name).
-          includes(:roles, access_groups: @group_associations).
-          page(params[:page]).per(25)
-      else
-        @users = User.active.order(:last_name, :first_name).includes(:roles, access_groups: @group_associations).page(params[:page]).per(25)
+      respond_to do |format|
+        format.html do
+          if params[:q].present?
+            @users = User.active.
+              text_search(params[:q]).
+              order(:last_name, :first_name).
+              includes(:roles, access_groups: @group_associations).
+              page(params[:page]).per(25)
+          else
+            @users = User.active.order(:last_name, :first_name).includes(:roles, access_groups: @group_associations).page(params[:page]).per(25)
+          end
+        end
+        format.xlsx do
+          @users = User.active.
+            order(:last_name, :first_name).
+            includes(:roles, access_groups: @group_associations)
+          date = Date.current.strftime('%Y-%m-%d')
+          filename = "user-permissions-#{date}.xlsx"
+          headers['Content-Disposition'] = "attachment; filename=#{filename}"
```

Comment:
This can be DRYed up quite a bit.  Set
```
@users = User.active.order(:last_name, :first_name).includes(:roles, access_groups: @group_associations)
```
outside of the if statement, then 
```
@users = @users.text_search(params[:q]) if params[:q].present?
```
and
```
@users = @users.page(params[:page]).per(25)
```
in the `format.html` block.

Reply:
✅ I like that code organization much better. I was trying to avoid the additional database calls, but I guess that's not critical in this case, since it's only a couple extra calls either way, and the code is much nicer.

Reply:
That actually won't call out to the DB until it's needed, so no extra DB calls.  Scopes only get resolved when you begin looping over the results.

Reply:
Very cool. I got psyched out by trying things in the console. This is much better.